x/bank/keeper/send.go

```diff
@@ -0,0 +1,262 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+// SendKeeper defines a module interface that facilitates the transfer of coins
+// between accounts without the possibility of creating coins.
+type SendKeeper interface {
+	ViewKeeper
+
+	InputOutputCoins(ctx sdk.Context, inputs []types.Input, outputs []types.Output) error
+	SendCoins(ctx sdk.Context, fromAddr sdk.AccAddress, toAddr sdk.AccAddress, amt sdk.Coins) error
+
+	SubtractCoins(ctx sdk.Context, addr sdk.AccAddress, amt sdk.Coins) (sdk.Coins, error)
+	AddCoins(ctx sdk.Context, addr sdk.AccAddress, amt sdk.Coins) (sdk.Coins, error)
+
+	SetBalance(ctx sdk.Context, addr sdk.AccAddress, balance sdk.Coin) error
+	SetBalances(ctx sdk.Context, addr sdk.AccAddress, balances sdk.Coins) error
```

Comment:
Glad you're getting to this.

In my mind these operations shouldn't exist independently of `Mint`/`Burn`. Whenever coins are added or subtracted supply should get tracked, not separately.

Is this something you're planning to address in this PR?

Just to note, I did a find usages on all of these methods and there is only one non-test usage (bank genesis). So if the desire is not to break tests, maybe there is a "test keeper" that allows this but for general module usage coins are minted or burned.

Reply:
> Is this something you're planning to address in this PR?

No, this PR will already be massive and keeping the error surface area must be kept to a minimum.

I'm not changing the functionality of the existing x/supply module. Please open an ADR if you want to discuss API/architecture improvements 👍

Reply:
Okay gotcha. Will do